### PR TITLE
Emit "VariableEmitted" on adding new variable in a test step

### PR DESCRIPTION
**Pull request by rihter007**

### CODE REVIEW

File: pkg/runner/test_runner.go [hunk 1]
```diff
@@ -136,7 +136,31 @@ func (tr *TestRunner) Run(
...
 	}
 
-	stepOutputs, err := newTestStepsVariables(t.TestStepsBundles)
+	emitterForStep := make(map[string]testevent.Emitter)
+	for _, sb := range t.TestStepsBundles {
+		emitterForStep[sb.TestStepLabel] = emitterFactory.New(sb.TestStepLabel)
+	}
+
+	stepOutputs, err := newTestStepsVariables(t.TestStepsBundles, func(tgtID string, stepLabel string, name string, value json.RawMessage) {
+		emitter := emitterForStep[stepLabel]
```
Comment (mimir-d) [inline pkg/runner/test_runner.go:145]: this callback ends up being called by multiple goroutines handling the steps/targets, right? this read should be sync'd somehow; it's not happening at the callsite in testStepsVariables.Add either (even if it's now just a readonly, there's no guarantee something might not change this map)

**Review by mimir-d — APPROVED** [review]

File: pkg/runner/test_runner.go (hunk 1 above)
Comment (rihter007) [reply pkg/runner/test_runner.go:145]: That is true, but all access is read only, so should be fine

Comment (mimir-d) [reply pkg/runner/test_runner.go:145]: note the thing i wrote in parens

Comment (rihter007) [reply pkg/runner/test_runner.go:145]: I agree, but currently the map and its usage is located nearby in terms of LoCs, so I would not overcomplicate for possible future safety

**Comment by rihter007** [comment]
Let's fix https://github.com/linuxboot/contest/issues/84 first

**Pull request merged**